pkg/publisher/fallback/publisher.go

```diff
@@ -0,0 +1,78 @@
+// fallback implements a publisher.Publisher that will try multiple other
+// Publishers in order until one suceeeds.
+//
+// The Publishers are tried in the order specified in the call to
+// NewFallbackPublisher. If and only if all of the Publishers return an error
+// result, the fallback publisher will also return an error result. Otherwise,
+// it will return the result of the first Publisher to succeed and will swallow
+// any errors. Subsequent Publishers will not be attempted after one succeeds.
+
+package fallback
+
+import (
+	"context"
+
+	"github.com/filecoin-project/bacalhau/pkg/model"
+	"github.com/filecoin-project/bacalhau/pkg/publisher"
+	"github.com/rs/zerolog/log"
+	"go.uber.org/multierr"
+)
+
+type fallbackPublisher struct {
+	publishers []publisher.Publisher
+}
+
+func NewFallbackPublisher(publishers ...publisher.Publisher) *fallbackPublisher {
+	return &fallbackPublisher{
+		publishers: publishers,
+	}
+}
+
+// fallback accepts a slice of publisher objects and passes them to the supplied
+// function in order, until one does not return an error value and then returns
+// that result. If all publishers return an error value, a composite error is
+// returned.
+func fallback[T any, P any](ctx context.Context, publishers []P, method func(P) (T, error)) (T, error) {
```

Comment:
ohhhh nice! generics - I'm assuming this won't introduce any problems with CI or compiling anything?

I'm sure the answer is no but felt compelled to comment because a part of my brain hasn't yet lept into "you can use generics now in go" territory :-)